darwin/dataset/download_manager.py

```diff
@@ -292,14 +297,39 @@ def _download_image(url: str, path: Path, api_key: str) -> None:
             response = requests.get(url, headers={"Authorization": f"ApiKey {api_key}"}, stream=True)
         # Correct status: download image
         if response.ok:
-            with open(str(path), "wb") as file:
-                for chunk in response:
-                    file.write(chunk)
+            if has_json_content_type(response):
+                _fetch_multiple_files(path, response)
+            else:
+                _write_file(path, response)
             return
         # Fatal-error status: fail
         if 400 <= response.status_code <= 499:
-            raise Exception(f"Request to ({url}) failed. Status code: {response.status_code}, content:\n{get_response_content(response)}.")
+            error = f"Request to ({url}) failed. Status code: {response.status_code}, content:\n{get_response_content(response)}."
+            print(error)
+            raise Exception(error)
         # Timeout
         if time.time() - start > TIMEOUT:
             raise Exception(f"Timeout url request ({url}) after {TIMEOUT} seconds.")
         time.sleep(1)
+
+
+def _fetch_multiple_files(path: Path, response: requests.Response) -> None:
+    urls = response.json()["urls"]
+    dir_path = Path(path).with_suffix("")
+    dir_path.mkdir(exist_ok=True, parents=True)
+    for url in urls:
+        filename = urllib.parse.urlparse(url).path.rsplit("/", 1)[-1]
+        path = dir_path / filename
+        response = requests.get(url, stream=True)
+        if response.ok:
+            _write_file(path, response)
+        else:
+            print(
```

Comment:
I don't think this is how we handle errors in darwin py in general. I think we raise exception instead and interrupt.

Reply:
Done